lib/pipelineTemplateVersionFactory.js

```diff
@@ -0,0 +1,153 @@
+     * @method createClass
+     * @param  {Object}     config               Template data
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.namespace   The template namespace
+     * @param  {String}     config.version       Version of the template
+     * @param  {String}     config.description   Description of the template
+     * @param  {String}     config.maintainer    Maintainer's email
+     * @param  {Object}     config.config        Config of the screwdriver-template.yaml
+     * @param  {String}     config.pipelineId    pipelineId of the template
+     * @return {PipelineTemplateVersionFactory}
+     */
+    createClass(config) {
+        return new PipelineTemplateVersionFactory(config);
+    }
+
+    /**
+     * Create a new template of the correct version (See schema definition)
+     * @method create
+     * @param  {Object}     config               Config object
+     * @param templateMetaFactory
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.namespace   The template namespace
+     * @param  {String}     config.version       Version of the template
+     * @param  {String}     config.description   Description of the template
+     * @param  {String}     config.maintainer    Maintainer's email
+     * @param  {Object}     config.config        Config of the screwdriver-template.yaml
+     * @param  {String}     config.pipelineId    pipelineId of the template
+     * @return {Promise}
+     */
+
+    create(config, templateMetaFactory) {
+        const createTime = new Date().toISOString();
+
+        return templateMetaFactory
+            .get({
+                name: config.name,
+                namespace: config.namespace
+            })
+            .then(pipelineTemplate => {
+                if (!pipelineTemplate) {
+                    return templateMetaFactory.create({
+                        pipelineId: config.pipelineId,
+                        namespace: config.namespace,
+                        name: config.name,
+                        maintainer: config.maintainer,
+                        createTime,
+                        updateTime: createTime
+                    });
+                }
+
+                return pipelineTemplate;
+            })
+            .then(pipelineTemplate => {
+                const isExactVersion = EXACT_VERSION_REGEX.exec(config.version);
+
+                if (isExactVersion) {
+                    return this.get({
+                        templateId: pipelineTemplate.id,
+                        version: config.version
+                    }).then(versionedConfig => {
+                        return { pipelineTemplate, versionedConfig };
```

Comment:
To be consistent, this step should return all the attributes `{ pipelineTemplate: ..., versionedConfig: ...,  isExactVersion: ...}` for all the scenarios
```suggestion
                        return { pipelineTemplate, versionedConfig, isExactVersion };
```